server/utils/match/index.ts

```diff
@@ -174,17 +175,19 @@ export const spaceBookingPersonNeedsSummaryCardRows = (
 
 export const occupancyViewSummaryListForMatchingDetails = (
   totalCapacity: number,
-  placementRequest: PlacementRequest,
+  placementRequest: PlacementRequestDetail,
   managerDetails: string,
 ): Array<SummaryListItem> => {
   const placementRequestDates = placementDates(placementRequest.expectedArrival, placementRequest.duration)
   const essentialCharacteristics = filterOutAPTypes(placementRequest.essentialCriteria)
+  const application = placementRequest.application as ApprovedPremisesApplication
```

Comment:
Does this need to check the `type`, or can we guarantee that this `Application` is always an `ApprovedPremisesApplication`

Reply:
I had a quick look into the BE code and saw that when we hit the `GET /placement-requests/{id}` endpoint, it essentially goes off and build the placement request including the application. To build the correct application type it executes this code:
<img width="2048" alt="Screenshot 2024-12-20 at 15 00 23 (2)" src="https://github.com/user-attachments/assets/e117fb60-1058-44c5-9bb5-59c5a6b0e8cb" />

So, the `is` stuff above determines which application type is attached to this placement request and builds it up accordingly. My understanding is that in this area (i.e. `Occupancy view`) the placements requests will all be of type `ApprovedPremisesApplicationEntity` as this `Occupancy view` area is functionality used for CAS1 applications. So, my conclusion is that, yes we can guarantee that this `Application` is always an `ApprovedPremisesApplication`.

Let me know if I'm missing anything :)